slatedb/src/admin.rs

```diff
@@ -179,6 +180,42 @@ impl Admin {
             .map_err(Into::<crate::Error>::into)
     }
 
+    /// Run the compactor in the foreground until the provided cancellation token is cancelled.
+    ///
+    /// This method blocks until `cancellation_token` is cancelled, at which point it requests a
+    /// graceful shutdown and waits for the compactor to stop.
+    pub async fn run_compactor(
+        &self,
+        cancellation_token: CancellationToken,
+    ) -> Result<(), crate::Error> {
+        let compactor = crate::CompactorBuilder::new(
+            self.path.clone(),
+            self.object_stores.store_of(ObjectStoreType::Main).clone(),
+        )
+        .with_system_clock(self.system_clock.clone())
+        .with_seed(self.rand.seed())
+        .build();
+
+        let mut run_task = tokio::spawn({
+            let compactor = compactor.clone();
```

Comment:
Not related to this PR, but I was wondering if the Inner pattern would make sense for Compactor. Seems like we could move all the state to a `CompactorInner` and let `Compactor` have just one Arc reference to it.

Reply:
Makes sense. I'm mostly just trying to feel out usage of rust idioms.

Reply:
🤣 me, too. me too.. 😛